plugin/core/clients.py

```diff
@@ -36,6 +37,26 @@ def get_window_env(window: sublime.Window, config: ClientConfig) -> 'Tuple[List[
     return expanded_args, env
 
 
+def get_expanding_variables(window: sublime.Window) -> dict:
+    variables = window.extract_variables()
+    variables.update({
+        "home": os.path.expanduser('~'),
+        "temp_dir": tempfile.gettempdir(),
+    })
+
+    return variables
+
+
+def expand_variables_for_dict(window: sublime.Window, dict_: dict) -> dict:
```

Comment:
NIT: I don't like the underscore suffix for a variable :(

Also, we can pin down what kind of dict this is, right? It must be a `Dict[str, Union[str, dict]]`. That should also be the return type if I read it correctly.

Reply:
> I don't like the underscore suffix for a variable :(

I dont either. But is using `dict` as a variable name a good practice? Or, what'a suggested one?

Reply:
> Also, we can pin down what kind of dict this is, right? It must be a Dict[str, Union[str, dict]]. That should also be the return type if I read it correctly.

If the target is not a more general `dict`, i.e., `Dict[str, Any], should we change the function name?

Reply:
>  It must be a Dict[str, Union[str, dict]]

What about `bool`, `int`, `float`?

```
"intelephense": {
    // ...
    "initializationOptions": {
        "clearCache": false, // <-------- bool
    },
},
```

Reply:
You're right. It's probably best to leave it at `Dic[str, Any]`.